dpbench/benchmarks/black_scholes/black_scholes_sycl_native_ext/black_scholes_sycl/_black_scholes_sycl.cpp

```diff
@@ -64,14 +64,22 @@ void black_scholes_sync(size_t /**/,
     if (!ensure_compatibility(price, strike, t, call, put))
         throw std::runtime_error("Input arrays are not acceptable.");
 
-    if (typenum != UAR_DOUBLE) {
-        throw std::runtime_error("Expected a double precision FP array.");
+    if (typenum == UAR_FLOAT) {
+        black_scholes_impl<float>(Queue, nopt, price.get_data<float>(),
+                                  strike.get_data<float>(), t.get_data<float>(),
+                                  rate, volatility, call.get_data<float>(),
+                                  put.get_data<float>());
+    }
+    else if (typenum == UAR_DOUBLE) {
+        black_scholes_impl<double>(
+            Queue, nopt, price.get_data<double>(), strike.get_data<double>(),
+            t.get_data<double>(), rate, volatility, call.get_data<double>(),
+            put.get_data<double>());
+    }
+    else {
+        throw std::runtime_error(
+            "Expected a double or single precision FP array.");
```

Comment:
It should be possible to directly bind functions with template parameters. [This ](https://pybind11.readthedocs.io/en/stable/advanced/functions.html#binding-functions-with-template-parameters)might be a neater solution. @oleksandr-pavlyk your thoughts?